### PR TITLE
[recommendation] - add manual tracing instrumentation

### LINKED ISSUE #59: Review and enhance tracing support for recommendation service (Python)
The following is a list of requirements that we need to evaluate before declaring v1 for trace telemetry. These requirements are across the entire application; Not all services will meet all requirements. Determine the relevant features for this service. 

- [ ] Automatic Instrumentation is being used where appropriate.
- [ ] Library instrumentation is being used if automatic instrumentation is unavailable.
- [ ] Services extend automatic instrumentation.
  - [ ] New attributes/events attached to existing spans.
  - [ ] New spans are being created from existing spans.
- [ ] Automatic and Manual Context Propagation are being demonstrated.
  - [ ] Interprocess context propagation is demonstrated.
- [ ] Telemetry is not being sampled upfront.
- [ ] Telemetry is not being filtered upfront.
- [ ] Baggage is being set and read appropriately (i.e., baggage must be explicitly set as attributes on spans)

Referencing: https://github.com/open-telemetry/opentelemetry-demo-webstore/issues/42

Service: https://github.com/open-telemetry/opentelemetry-demo-webstore/blob/main/src/recommendationservice/README.md

### PR DESCRIPTION
**Pull request by puckpuck**
Fixes #59 
- [x] Services extend automatic instrumentation.
  - [x] New attributes/events attached to existing spans.
  - [x] New spans are being created from existing spans.

## Changes

Adds an attribute to the auto-instrumented span, and moves logic to get the list of products to a new function wrapped in a new span with attributes.

### CODE REVIEW

File: src/recommendationservice/recommendation_server.py [hunk 1]
```diff
@@ -66,12 +59,34 @@ def Watch(self, request, context):
...
             status=health_pb2.HealthCheckResponse.UNIMPLEMENTED)
 
+
+def get_product_list(request_product_ids):
+    with tracer.start_as_current_span("get_product_list") as span:
+        max_responses = 5
+        # fetch list of products from product catalog stub
+        cat_response = product_catalog_stub.ListProducts(demo_pb2.Empty())
+        product_ids = [x.id for x in cat_response.products]
+        span.set_attribute("app.products.count", len(product_ids))
```
Comment (cartersocha) [inline src/recommendationservice/recommendation_server.py:69]: Probably want to hold off on merging this until we resolve the service name inclusion / attribute naming conventions in the other pr

File: src/recommendationservice/recommendation_server.py [hunk 2]
```diff
@@ -37,21 +37,14 @@
...
-        cat_response = product_catalog_stub.ListProducts(demo_pb2.Empty())
-        product_ids = [x.id for x in cat_response.products]
-        filtered_products = list(set(product_ids)-set(request.product_ids))
-        num_products = len(filtered_products)
-        num_return = min(max_responses, num_products)
-        # sample list of indicies to return
-        indices = random.sample(range(num_products), num_return)
-        # fetch product ids from indices
-        prod_list = [filtered_products[i] for i in indices]
+        prod_list = get_product_list(request.product_ids)
```
[the remark this answers is not in the document]
Comment (puckpuck) [reply src/recommendationservice/recommendation_server.py:45]: The recommendation service will recommend products that are not currently on the user's page (part of the request).

[the remark this answers is not in the document]
Comment (mic-max) [reply src/recommendationservice/recommendation_server.py:45]: Ahh gotcha that makes a lot of sense 😅

**Review by julianocosta89 — APPROVED** [review]
🚀

**Pull request merged**